client/app/account/billing/payment/method/add/billing-payment-method-add.controller.js

```diff
@@ -178,20 +186,20 @@ export default class BillingPaymentMethodAddCtrl {
       } else {
         _.set(addParams, 'billingContactId', paymentMethodContact.id);
       }
+    }
 
-      if (!this.model.selectedPaymentMethodType.original) {
-        const isQueryParamsInHash = this.$window.location.hash.indexOf('?') > 0;
-        addParams = _.merge(addParams, {
-          register: true,
-          callbackUrl: {
-            cancel: [this.$window.location.href, 'status=cancel'].join(isQueryParamsInHash ? '&' : '?'),
-            error: [this.$window.location.href, 'status=error'].join(isQueryParamsInHash ? '&' : '?'),
-            failure: [this.$window.location.href, 'status=failure'].join(isQueryParamsInHash ? '&' : '?'),
-            pending: [this.$window.location.href, 'status=pending'].join(isQueryParamsInHash ? '&' : '?'),
-            success: [this.$window.location.href, 'status=success'].join(isQueryParamsInHash ? '&' : '?'),
-          },
-        });
-      }
+    if (!this.model.selectedPaymentMethodType.original) {
+      const isQueryParamsInHash = this.$window.location.hash.indexOf('?') > 0;
+      addParams = _.merge(addParams, {
+        register: true,
+        callbackUrl: {
+          cancel: [this.$window.location.href, 'status=cancel'].join(isQueryParamsInHash ? '&' : '?'),
```

Comment:
Not yours, but this could be improved with a templated string `status=${status}` and maybe call a function to generate callbackUrl ?